bin/BuildPackages.sh

```diff
@@ -186,8 +208,33 @@ do
         ;;
       esac;
       ) || build_fail
+}
+
+      (buildpackage > >(tee $LOGDIR/$dir.out) 2> >(tee $LOGDIR/$dir.err >&2) )> >(tee $LOGDIR/$dir.log ) 2>&1
       # end subshell
+      # remove superfluous log files if there was no error message
+      if [ ! -s "$LOGDIR/$dir.err" ]
+        then
+        rm -f "$LOGDIR/$dir.err"
+        rm -f "$LOGDIR/$dir.out"
+      fi
     else
       echo "$dir is not a GAP package -- no PackageInfo.g"
-    fi;
-done;
+    fi
+done < <(find -L ./ -maxdepth 1 -mindepth 1 -type d | sed 's|./||')
```

Comment:
But that advice only applies to reading from a file, and you are not reading from a file. You are iterating over a list of paths, and you definitely should use a `for` loop for that. (Feel free to use `$(...)$` instead of backticks, though).